src/components/app/ListOfPublishedProfiles.js

```diff
@@ -0,0 +1,155 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import React, { PureComponent } from 'react';
+import memoize from 'memoize-immutable';
+import { AppHeader } from './AppHeader';
+
+import {
+  listAllProfileData,
+  type ProfileData,
+} from 'firefox-profiler/app-logic/published-profiles-store';
+import { formatSeconds } from 'firefox-profiler/utils/format-numbers';
+import { ProfileMetainfoSummary } from 'firefox-profiler/components/shared/ProfileMetainfoSummary';
+
+import type { Milliseconds, StartEndRange } from 'firefox-profiler/types/units';
+
+import './ListOfPublishedProfiles.css';
+
+// This component displays all published profile, and makes it possible to load
+// them by clicking on them, or delete them.
+
+const ONE_DAY_IN_MS = 24 * 60 * 60 * 1000;
+const ONE_YEAR_IN_MS = 365 * ONE_DAY_IN_MS;
+const dateFormats = {
+  thisDay: memoize(
```

Comment:
Initially I was confused by the memoization, it seems like it's only to make the DateTimeFormat lazy initialized. Please add a comment here to make it more clear. I would have probably used `let _ancient: DateTimeFormat?;` and then initialize it where it's used, but this is fine too.